File: scripts/benchmark.js
const fftBenchmark = function (nIter, testFunctions, testVectors, testSizes, now) {
  var report = {'nIter': nIter}
  testFunctions.forEach((f, idx) => {
    report[f.name] = {}
    testSizes.forEach(n => {
      var x = testVectors[n]['in']
      var t0 = now()
      for (var i=0; i<nIter; i++) {
        f(x)
      }
      var delta = now() - t0
      report[f.name][n] = delta
    })
  })
  return report
}

const format = function (val, n) {
  if (n === undefined) {
    n = 4
  }
  return val.toFixed(n)
}


const formatReport = function (report) {
  var nIter = report['nIter']
  Object.keys(report).forEach(name => {
    Object.keys(report[name]).forEach(n => {
      var delta = report[name][n] / 1000
      console.log(`n=${n}, ${name}: ${format(delta)} sec, ${format(delta/nIter, 7)} per loop, ${format(nIter / delta, 2)} iter/sec`)
    })
  })
}

if (typeof require != 'undefined') {
  exports.fftBenchmark = fftBenchmark
  exports.formatReport = formatReport
} else {
  window.fftBenchmark = fftBenchmark
  window.formatReport = formatReport
}

File: scripts/benchmark.py
import time
import json

import numpy as np


def main():
    # n = 8192
    n_iter = 2000
    sizes = [32768]
    sizes = [512, 2048]
    for n in sizes:
        a = np.random.rand(n) + 1j*np.random.rand(n)
        # a = np.arange(n) + 1j*np.arange(n)
        t0 = time.time()
        for i in range(n_iter):
            np.fft.fft(a)
        delta = time.time() - t0
        print(f"size: {n}: {delta:.7f} sec, {delta/n_iter:.7f} per loop, {n_iter/delta:.7f} iter per second")


if __name__ == '__main__':
    main()
